config/crds/awsprovider_v1alpha1_awsclusterproviderspec.yaml

```diff
@@ -33,6 +33,301 @@ spec:
           - cert
           - key
           type: object
+        clusterConfiguration:
```

Comment:
Assuming this is coming from kubeadm, and we'll have to keep this updated as and when we adopt new kubeadm version, correct?

Reply:
Same question for `InitConfiguration` and `JoinConfiguration`

Reply:
There is also an outstanding todo for figuring out how to handle the kubeadm config for interacting with multiple versions of kubeadm. But yes, updates will be needed to support newer versions.

Reply:
I'm working on https://github.com/kubernetes-sigs/cluster-api/issues/730 that will address update issues. For now I think this will be good enough for a v1alpha1 release